draft-nro-bulk-rdap.md

```diff
@@ -48,184 +48,294 @@ feature of this service.
 
 # Bulk Data Format {#bulk_data_format}
 
-The data returned for a Bulk RDAP request ((#bulk_rdap_url)) is a JSON object comprising JSON members for metadata and
-JSON data for a single RDAP object class.
+The data returned for a Bulk RDAP request ((#bulk_rdap_url)) is a sequence of JSON objects, each followed by a newline
+('\n') character. Such a sequence is commonly known as [JSON Lines](https://jsonlines.org/), or newline-delimited JSON,
+and makes bulk data streaming computationally efficient for the clients.
+The first object in the returned sequence provides the metadata information for the bulk data, and the following objects
+are data objects for one or more RDAP object classes ([@!RFC9083, section 5]).
 
 ## Metadata
 
-The following JSON members for metadata MUST be included:
-
-* "rdapConformance" -- An array of strings ([@!RFC9083, section 4.1]) to signal the RDAP extensions the data is based
-  on.
-* "serial" -- An unsigned 32-bit number for sequencing the data snapshots. It is RECOMMENDED to follow the serial
-  number-wrapping arithmetic from [@!RFC1982].
-
-## Data for an Object Class
-
-The JSON data for an RDAP object class is an "objects" member -- a JSON array comprising all the objects for that class.
-It is RECOMMENDED that an RIR provide bulk data for IP Network, Autonomous System Number, Domain, Nameserver, and Entity
-object classes ([@!RFC9083, section 5]).
-
-Furthermore, within the JSON array:
-
-* The "self" links for each primary object and the secondary objects it contains MUST be included to have parity with
-  the "ref" element in the Bulk Whois.
-* For compactness, it is NOT RECOMMENDED to include details for a secondary object beside its "self" link, "handle" when
-  defined, and relationship to the primary object (e.g., using the "roles" member in an Entity object to relate to an IP
-  Network object).
-* An "rdapConformance" member MAY be included at the top of a primary object if the RDAP extensions used to produce it
-  are different from those listed in the "rdapConformance" member for metadata.
-
-## Example
-
-Here is an elided example of a JSON object containing bulk data for the IP Network object class:
+The metadata JSON object MUST include the following members:
+
+* "extensionId" -- the "nroBulkRdap1" string (see (#rdap_extensions_registry))
```

Comment:
Why not use `rdapConformance` here?

Reply:
I considered that but would there be any other extension mentioned in the metadata object, beside the one for the Bulk RDAP? Though I agree that `rdapConformance` gives us a greater flexibility if we ever need to have more than one extension mentioned in the metadata.

Reply:
I didn't know about the requirement from section 4.1, that's interesting.  If clients have been written on the basis that the top level will always have an `rdapConformance` member, then omitting that here could be an issue, whereas including it in nested objects is less likely to cause problems, even though it's not in keeping with the text.  Given how searches are handled today, and their being the most analogous existing response, I think including `rdapConformance` at the top level with all of the relevant conformance codes, and then adding nested `rdapConformance` members for each individual object, would be the approach most consistent with how things have gone before, without sacrificing any detail about the specific extensions involved in the construction of each object.

Reply:
`rdapConformance` in the metadata object will have the union of all of the conformance codes used in each individual object, but otherwise yep, that's right.

Reply:
OK